pkgs/applications/misc/bloodhound/default.nix

```diff
@@ -0,0 +1,127 @@
+{ lib
+, stdenv
+, fetchurl
+, makeWrapper
+, alsa-lib
+, at-spi2-atk
+, at-spi2-core
+, atk
+, cairo
+, cups
+, dbus
+, expat
+, fontconfig
+, freetype
+, gdk-pixbuf
+, glib
+, gnome2
+, gtk3
+, libGL
+, libappindicator-gtk3
+, libdrm
+, libnotify
+, libpulseaudio
+, libuuid
+, libxcb
+, libxkbcommon
+, libxshmfence
+, mesa
+, nspr
+, nss
+, pango
+, systemd
+, unzip
+, xdg-utils
+, xorg
+}:
+stdenv.mkDerivation rec {
+  pname = "bloodhound";
+  binaryName = "BloodHound";
+  version = "4.3.1";
+  src = fetchurl {
+    url = "https://github.com/BloodHoundAD/BloodHound/releases/download/v${version}/BloodHound-linux-x64.zip";
+    sha256 = "sha256-OtQNrbPGAw5WoDeli6+REcx9ajor0CeF89L7Gg44aB0=";
+  };
+
+  rpath = lib.makeLibraryPath [
+    alsa-lib
+    at-spi2-atk
+    at-spi2-core
+    atk
+    cairo
+    cups
+    dbus
+    expat
+    fontconfig
+    freetype
+    gdk-pixbuf
+    glib
+    gnome2.GConf
+    gtk3
+    libGL
+    libappindicator-gtk3
+    libdrm
+    libnotify
+    libpulseaudio
+    libuuid
+    libxcb
+    libxkbcommon
+    mesa
+    nspr
+    nss
+    pango
+    systemd
+    xorg.libX11
+    xorg.libXScrnSaver
+    xorg.libXcomposite
+    xorg.libXcursor
+    xorg.libXdamage
+    xorg.libXext
+    xorg.libXfixes
+    xorg.libXi
+    xorg.libXrandr
+    xorg.libXrender
+    xorg.libXtst
+    xorg.libxkbfile
+    xorg.libxshmfence
+  ] + ":${stdenv.cc.cc.lib}/lib64";
+
+  buildInputs = [
+    gtk3 # needed for GSETTINGS_SCHEMAS_PATH
+  ];
+
+  nativeBuildInputs = [ makeWrapper unzip ];
+
+  dontUnpack = true;
+  dontBuild = true;
+  dontPatchELF = true;
+
+  installPhase = ''
+    runHook preInstall
+
+    unzip $src
+    mkdir -p $out
+    mkdir -p $out/{bin,lib/${binaryName}}
+    mv BloodHound-linux-x64/* $out/lib/${binaryName}
+    chmod +x $out/lib/${binaryName}/${binaryName}
+
+    patchelf --set-interpreter ${stdenv.cc.bintools.dynamicLinker} \
+       $out/lib/${binaryName}/${binaryName}
+
+    makeWrapper $out/lib/${binaryName}/${binaryName} $out/bin/${binaryName} \
+      --prefix XDG_DATA_DIRS : $GSETTINGS_SCHEMAS_PATH \
+      --prefix PATH : ${lib.makeBinPath [xdg-utils]} \
+      --prefix LD_LIBRARY_PATH : ${rpath}
```

Comment:
Can we avoid this and use patchelf instead?

Reply:
This is my first time contribution, I'd guess other packages avoid this as well, do you have any reference package to recommend me ?

I'll start to have a look at others packages to see how it's done, once patchelf is used,  `LD_LIBRARY_PATH` will no longer be needed right ?
Will the `PATH` variable for `xdg-utils` won't be needed ?

Reply:
patchelf replaces LD_LIBRARY_PATH when used correctly but the modification to PATH is still required

Reply:
patchelf needs to be used with `--set-rpath` and it should be good enough, like so : 
```nix
    patchelf --set-interpreter ${stdenv.cc.bintools.dynamicLinker} \
       $out/lib/BloodHound/BloodHound --set-rpath ${finalAttrs.rpath}
```
where `finalAttrs.rpath` is defined as `lib.makeLibaryPath [<all_missing_libraries>]` ?